---
title: ""
author: ""
date: ""
output:
  pdf_document:
    toc: false
    latex_engine: xelatex
header-includes:
  \usepackage{fontspec}
  \usepackage[utf8]{inputenc}
  \setmainfont{Arial}
  \usepackage{pdfpages}
  \usepackage{graphicx}
  \usepackage{siunitx}
params:
  args: ''
---

```{r read_arg, include = F}
args <-  params$args
eval <- args[1]
```

```{r setup, include=FALSE, message=FALSE, echo=FALSE, eval = T}
library(knitr)
library(dplyr)
library(tidyverse)
library(ggplot2)
library(patchwork)
library(pROC)
library(gtsummary)
library(ggpubr)
library(here)
library(gt)
library(ggnewscale)
library(ggrepel)
library(ComplexHeatmap)
library(ggtext)
knitr::opts_chunk$set(message = F,
                      warning = F,
                      echo = F,
                      eval = eval)
# # 
# # Colour palette
cols <- MetBrewer::met.brewer("Hokusai1", n = 9)
cols2 <- MetBrewer::met.brewer("Hiroshige", n = 9)
cols3 <- colorRampPalette(cols2)(16)

# 
# # Themes
theme_gtsummary_journal(journal = "nejm")
theme_gtsummary_compact()

# # here
here::i_am("2-markdown/Supplementary_WID_SMK.Rmd")

# # Functions
source(here("0-source/correct_ic.R"))
source(here("0-source/smallAUCplot.R"))
source(here("0-source/manhattan.R"))
source(here("0-source/qqplot.R"))
source(here("0-source/plot_3_rocs.R"))
source(here("0-source/plot_2_rocs.R"))
```


\vspace*{4cm}


\large 
\textbf{Supporting information for}\newline
Cigarette smoking and e-cigarette use induce shared DNA methylation changes linked to carcinogenesis
\vspace{1em}

\normalsize

Chiara Herzog, Allison Jones, Iona Evans, Janhavi R. Raut, Michal Zikan, David Cibula, Andrew Wong, Hermann Brenner, Rebecca C. Richmond, and Martin Widschwendter\newline

Martin Widschwendter\newline
email: martin.widschwender@uibk.ac.at

\vspace{1.5em}

\textbf{This PDF file includes:}

> Figures S1 to S13 \newline
> Tables S1 to S11\newline
> Information for Supplementary Movie 1

\newpage

\tableofcontents
\newcommand\invisiblesection[1]{%
  \refstepcounter{section}%
  \addcontentsline{toc}{section}{\protect\numberline{\thesection}#1}%
  \sectionmark{#1}}
  
\newpage
  
\section{Supplementary Figures}

\invisiblesection{Supplementary Figure 1. Analysis workflow overview.}

\includegraphics[]{1-figure-panels/s1_analysis-workflow.pdf}

\textbf{Supplementary Figure 1. Analysis workflow overview.}

\newpage

\invisiblesection{Supplementary Figure 2. Manhattan and qq-plots for smoking-associated CpGs in buccal samples.}

```{r s2-4-manh-qq-plots}
# Need to be manually assembled
# Buccal -------------
lowmi <- read.csv(here("0-source/lowMI_names.csv"), header = F)
notonepicv2 <- read.csv(here("0-source/notonEPICv2_names.csv"), header = F)
db_buccal <- readRDS(here("1-analysis-pipeline/1-delta-beta/1a-buccal/delta-beta.Rds")) |>
  as.data.frame() |>
  tibble::rownames_to_column("cg") |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  mutate(padj = p.adjust(type, method = "holm"),
         sig = ifelse(padj < 0.05, "yes", "no"))

db <- db_buccal
manhplot <- manhattan(db, sample = "buccal")
ggsave(plot = manhplot,
       filename = here("2-markdown/1-figure-panels/ewas-buccal.png"), width = 10, height = 4)
qqplot(db_buccal$type, filename = here("2-markdown/1-figure-panels/qq-buccal.png"))

# Cervical -------------
db_cervical <- readRDS(here("1-analysis-pipeline/1-delta-beta/1b-cervical/delta-beta.Rds")) |>
  as.data.frame() |>
  tibble::rownames_to_column("cg") |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  mutate(padj = p.adjust(type, method = "holm"),
         sig = ifelse(padj < 0.05, "yes", "no"))

db <- db_cervical
manhplot <- manhattan(db, sample = "cervical")
ggsave(plot = manhplot,
       filename = here("2-markdown/1-figure-panels/ewas-cervical.png"), width = 10, height = 4)
qqplot(db_cervical$type, filename = here("2-markdown/1-figure-panels/qq-cervical.png"))

# Blood -------------
db_blood <- readRDS(here("1-analysis-pipeline/1-delta-beta/1c-blood/delta-beta.Rds")) |>
  as.data.frame() |>
  tibble::rownames_to_column("cg") |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  mutate(padj = p.adjust(type, method = "holm"),
         sig = ifelse(padj < 0.05, "yes", "no"))

db <- db_blood
manhplot <- manhattan(db, sample = "blood")
ggsave(plot = manhplot,
       filename = here("2-markdown/1-figure-panels/ewas-blood.png"), width = 10, height = 4)
qqplot(db_blood$type, filename = here("2-markdown/1-figure-panels/qq-blood.png"))


# Histograms
db_buccal <- readRDS(here("1-analysis-pipeline/1-delta-beta/1a-buccal/delta-beta.Rds")) |>
  as.data.frame() |>
  tibble::rownames_to_column("cg") |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  mutate(padj = p.adjust(type, method = "holm"),
         sig = ifelse(padj < 0.05, "yes", "no")) |>
  tidyr::pivot_longer(db_epithelial:db_immune,
                      values_to = "beta",
                      names_to = "dbtype")

histplot <- db_buccal |>
  ggplot(aes(x = beta,
             fill = dbtype)) +
  geom_histogram(alpha = 0.7,
                 bins = 200) +
  theme_minimal() +
  labs(x = "∆β",
       y = "n") +
  theme(
    legend.position = "top",
    aspect.ratio = 1
    ) +
  scale_fill_manual(values = cols[c(3, 5)],
                     name = "",
                     labels = c("epithelial ∆β",
                                "immune ∆β"))
ggsave(plot = histplot,
       filename = here("2-markdown/1-figure-panels/histplot-buccal.png"), width = 4, height = 4)

db_cervical <- readRDS(here("1-analysis-pipeline/1-delta-beta/1b-cervical/delta-beta.Rds")) |>
  as.data.frame() |>
  tibble::rownames_to_column("cg") |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  mutate(padj = p.adjust(type, method = "holm"),
         sig = ifelse(padj < 0.05, "yes", "no")) |>
  tidyr::pivot_longer(db_epithelial:db_immune,
                      values_to = "beta",
                      names_to = "dbtype")

histplot <- db_cervical |>
  ggplot(aes(x = beta,
             fill = dbtype)) +
  geom_histogram(alpha = 0.7,
                 bins = 200) +
  theme_minimal() +
  labs(x = "∆β",
       y = "n") +
  theme(
    legend.position = "top",
    aspect.ratio = 1
    ) +
  scale_fill_manual(values = cols[c(3, 5)],
                     name = "",
                     labels = c("epithelial ∆β",
                                "immune ∆β"))
ggsave(plot = histplot,
       filename = here("2-markdown/1-figure-panels/histplot-cervical.png"), width = 4, height = 4)


db_blood <- readRDS(here("1-analysis-pipeline/1-delta-beta/1c-blood/delta-beta.Rds")) |>
  as.data.frame() |>
  tibble::rownames_to_column("cg") |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  mutate(padj = p.adjust(type, method = "holm"),
         sig = ifelse(padj < 0.05, "yes", "no")) |>
  dplyr::filter(!cg %in% c(notonepicv2$V1,lowmi$V1)) |>
  tidyr::pivot_longer(db_lymphoid:db_myeloid,
                      values_to = "beta",
                      names_to = "dbtype")

histplot <- db_blood |>
  ggplot(aes(x = beta,
             fill = dbtype)) +
  geom_histogram(alpha = 0.7,
                 bins = 200) +
  theme_minimal() +
  labs(x = "∆β",
       y = "n") +
  theme(
    legend.position = "top",
    aspect.ratio = 1
    ) +
  scale_fill_manual(values = cols[c(3, 5)],
                     name = "",
                     labels = c("lymphoid ∆β",
                                "myeloid ∆β"))
ggsave(plot = histplot,
       filename = here("2-markdown/1-figure-panels/histplot-blood.png"), width = 4, height = 4)
```

\includegraphics[trim = 0 11cm 0 0]{1-figure-panels/s2_ewas-buccal.pdf}

\textbf{Supplementary Figure 2. Manhattan and qq-plots for smoking-associated CpGs in buccal samples after accounting for age and immune cell proportion. a} Manhattan plot for smoking EWAS in buccal samples. CpGs were considered significant if they passed Bonferroni correction (equivalent to p \textless 8.2e-08). \textbf{b} qq plot for expected and observed p values in buccal sample EWAS. \textbf{c} delta beta (∆ β) values by epithelial and immune fraction in buccal samples.

\newpage

\invisiblesection{Supplementary Figure 3. Manhattan and qq-plots for smoking-associated CpGs in blood samples.}

\includegraphics[trim = 0 11cm 0 0]{1-figure-panels/s3_ewas-blood.pdf}

\textbf{Supplementary Figure 3. Manhattan and qq-plots for smoking-associated CpGs in blood samples after accounting for age and myeloid cell proportion. a} Manhattan plot for smoking EWAS in blood samples. CpGs were considered significant if they passed Bonferroni correction (equivalent to p \textless 7.9e-08). \textbf{b} qq-plot for expected and observed p values in blood sample EWAS. \textbf{c} delta beta (∆ β) values by lymphoid and myeloid fraction in blood samples.

\newpage

\invisiblesection{Supplementary Figure 4. Manhattan and qq-plots for smoking-associated CpGs in cervical samples.}

\includegraphics[trim = 0 11cm 0 0]{1-figure-panels/s4_ewas-cervical.pdf}

\textbf{Supplementary Figure 4. Manhattan and qq-plots for smoking-associated CpGs in cervical samples after accounting for age and immune cell proportion. a} Manhattan plot for smoking EWAS in cervical samples. CpGs were considered significant if they passed Bonferroni correction (equivalent to p \textless 7.9e-08). \textbf{b} q-q plot for expected and observed p values in cervical sample EWAS. \textbf{c} delta beta (∆ β) values by epithelial and immune fraction in cervical samples.

\newpage

\invisiblesection{Supplementary Figure 5. Clustering results for cell-specific delta-beta values of significant CpGs using two different approaches.}

```{r s5-clustering}
load(here("1-analysis-pipeline/2-output/WID_SMK_cpgs.Rdata"))

WID_SMK_cpgs <- WID_SMK_cpgs |>
  dplyr::mutate(set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                             "immune hypoM",
                                             "distal epithelial hyperM",
                                             "proximal epithelial hyperM")))
a <- WID_SMK_cpgs |>
  ggplot(aes(x = umap1,
             y = umap2,
             colour = set)) +
  geom_point() +
  theme_classic()+
  theme(legend.position = "top",
        legend.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank(),
        axis.line = element_blank()) +
  annotate("segment", x = -7, xend = -7, y = -7, yend = -4.5,
           linewidth = 0.5, arrow = arrow(length = unit(0.3, "cm"))) +
  annotate("segment", x = -7, xend = -5, y = -7, yend = -7,
           linewidth = 0.5, arrow = arrow(length = unit(0.3, "cm"))) +
  annotate("text", x = -7, y = -2, label = "UMAP 2", angle = 90, size = 3.5) +
  annotate("text", x = -3, y = -7, label = "UMAP 1", size = 3.5)  +
  xlab("") +
  ylab("") +
  scale_colour_manual(values = cols[c(9, 6, 3, 1)]) +
  guides(colour = guide_legend(nrow = 2, byrow = T))

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s5a.pdf"), width = 5, height = 5)
print(a)
dev.off()

# Additional clustering approach (delta-beta matrix)
load(here("1-analysis-pipeline/2-output/plot_matrix.Rdata"))
colnames(mat) <- c('buccal\n∆β epithelial',
                   'buccal\n∆β immune',
                   'cervical\n∆β epithelial',
                   'cervical\n∆β immune',
                   'blood\n∆β myeloid',
                   'blood\n∆β lymphoid')

ha = columnAnnotation(
  exposure = c("proximal", "distal", rep("distal", 3), 'proximal'),
  cell = c('epithelial', 'epithelial', rep('immune',4)),
  col = list(exposure = c("proximal" = cols[9],
                          "distal" = cols[7]),
             cell = c("epithelial" = cols[3],
                        "immune" = cols[6])),
  simple_anno_size = unit(0.3, "cm"),
  show_annotation_name = F
  )

b<- ComplexHeatmap::Heatmap(mat,
                        clustering_distance_rows = 'manhattan',
                        clustering_method_rows = 'ward.D',
                        row_km = 4,show_row_names = F,
                        name = "∆ β",
                        col = circlize::colorRamp2(breaks = seq(from = -0.2, to = 0.2,
                                                                        length.out = 9),
                                                           colors = rev(cols2))
                        )

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s5b.pdf"), width = 4, height = 8)
print(b)
dev.off()

```

\includegraphics{1-figure-panels/s5_clustering.pdf}

\textbf{Supplementary Figure 5. Clustering results for cell-specific delta-beta values of significant CpGs using two different approaches. a} Uniform manifold approximation and projection (UMAP) of the delta-beta of CpGs significantly associated with smoking in at least one tissue indicates the existence of four clusters. Clusters identified here are visualised in Figure 1c and named based on differential methylation in specific tissues and cell types. \textbf{b} A second, independent distance-based clustering, utilizing Manhattan distance and Ward's D based on the delta-beta matrix of CpGs significantly associated with smoking in at least one tissue reveals similar clustering as shown in textbf{a} (and Figure 1c).

\newpage

\invisiblesection{Supplementary Figure 6. Polycomb group target enrichment in CpGs in the four sets, association with expression, and overlap with previously identified sites.}

```{r s6-pcgt}
# a: PCGT enrichment ---------------------------
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

# Load single occupancy CpGs - previously identified, based on Lee et al. 2006 publication; single occupancy in TSS200 by either EZH, SUZ12,
load(here("0-source/single_occupancy.Rdata"))
source(here("0-source/enrich.R"))
load(here("1-analysis-pipeline/3-output/venn-sets.Rdata"))

# Gene universe: genes that are associated in genes at intersect of delta betas
db_buccal <- readRDS(here("1-analysis-pipeline/1-delta-beta/1a-buccal/delta-beta.Rds"))
db_blood <- readRDS(here("1-analysis-pipeline/1-delta-beta/1c-blood/delta-beta.Rds"))
db_cervical <- readRDS(here("1-analysis-pipeline/1-delta-beta/1b-cervical/delta-beta.Rds"))
lowmi <- read.csv(here("0-source/lowMI_names.csv"), header = F)
notonepicv2 <- read.csv(here("0-source/notonEPICv2_names.csv"), header = F)
cpgs <- intersect(rownames(db_buccal), intersect(rownames(db_blood), rownames(db_cervical)))
rm(db_buccal, db_blood, db_cervical);gc()
cpgs <- cpgs[!cpgs %in% c(notonepicv2$V1, lowmi$V1)]

genes <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19, lociNames = cpgs)
genes <- unique(as.character(stringr::str_split(genes$UCSC_RefGene_Name, ";", simplify = T)))
genes <- genes[genes !=""]

df <- enrich(sets, pcgt_single, genes)
set <- names(sets)

ext6a <- df |> ggplot(aes(x = set,
                      y = log(or),
                      colour = set)) +
  annotate("rect",
           xmin = c(0.5, 2.5), xmax = c(1.5, 3.5), ymin = c(-Inf, -Inf), ymax = c(Inf, Inf),
           fill = "lightgray",
           alpha = 0.5) +
  geom_linerange(aes(ymin = log(ci_lo),
                     ymax = log(ci_hi))) +
  geom_point(shape = 18,
             size = 5) +
  coord_flip() +
  scale_x_discrete(limits = rev(set),
                   labels = rev(set)) +
  geom_hline(yintercept = 0,
             linetype = "dashed") +
  theme_minimal() +
  theme(panel.grid = element_blank())+
  ylab("Enrichment for PCGT genes\nlog(odds ratio)") +
  xlab("") +
  theme(legend.position = "none") +
  scale_colour_manual(values = cols[c(3, 9, 6, 1)])

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext6a.pdf"), width = 4, height = 5)
print(ext6a)
dev.off()
# 
# # b: expression ---------------------------
load(here("1-analysis-pipeline", "8-output", "expression.Rdata"))

ext6b <- df |> 
  ggplot(aes(x = p)) +
  geom_histogram(bins = 15) +
  facet_wrap(~region) +
  theme_bw() +
  labs(x = 'correlation p value', y = 'count') +
  theme(aspect.ratio = 1)


grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext6b.pdf"), width = 6.5, height = 5)
print(ext6b)
dev.off()

# c: expression ---------------------------
load(here("1-analysis-pipeline", "8-output", "expression.Rdata"))

ext6c <- df |> 
  dplyr::mutate(padj = p.adjust(p)) |> 
  dplyr::filter(padj < 0.05) |> 
  ggplot(aes(x = cor)) +
  geom_density(aes(fill = region),
               alpha = 0.5,
               bw = 0.06) +
  geom_vline(xintercept = 0,
             linetype = 'dashed') +
  theme_bw() +
  labs(x = 'correlation with expression', y = 'density') +
  theme(aspect.ratio = 1) +
  scale_fill_manual(values = cols2[c(1, 2, 3, 5, 7, 8)])


grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext6c.pdf"), width = 6.5, height = 5)
print(ext6c)
dev.off()

# # d: previous studies ---------------------------
load(here("1-analysis-pipeline/2-output/WID_SMK_cpgs.Rdata"))
WID_SMK_cpgs <- WID_SMK_cpgs |>
  dplyr::filter(chr != "chrX")

# Joehanes
ewas <- readxl::read_xlsx(here("0-source/joehanes-suppl.xlsx"), sheet = 3, skip = 2) |>
  janitor::clean_names()
probes <- data.frame(source = rep("Joehanes", length(ewas$probe_id)),
                     cg = ewas$probe_id)
# Teschendorff
tx <- pdftools::pdf_text(here("0-source/jamaonc-2015.pdf"))
tmp <- unlist(stringr::str_split(tx[25:111], "[\\r\\n]+")) # separating lines
tmp1 <- stringr::str_split(tmp, "   ") # splitting
# sum(grepl("cg", tmp1)) # 1501 CpGs as reported in paper
cgs <- unlist(tmp1)[grepl("cg", unlist(tmp1))]
# only new:
cgs <- cgs[!cgs %in% probes$cg]
tmp <- data.frame(source = rep("Teschendorff", length(cgs)),
                  cg = cgs)
probes <- rbind(probes, tmp)

# Christiansen
christ <- readxl::read_xlsx(here("0-source/christiansen_13148_2021_1018_MOESM2_ESM.xlsx"), sheet = 2, skip = 1) |>
  dplyr::filter(!IlmnID %in% probes$cg)
tmp <- data.frame(source = rep("Christiansen", length(christ$IlmnID)),
                  cg = christ$IlmnID)
probes <- rbind(probes, tmp)
load(here("1-analysis-pipeline/2-output/WID_SMK_cpgs.Rdata"))
tmp <- WID_SMK_cpgs |>
  dplyr::left_join(probes)

x <- table(tmp$set, tmp$source, useNA = "ifany")
x <- as.data.frame(x) |>
  dplyr::mutate(Var2 = ifelse(is.na(Var2), "This study", as.character(Var2)))

data <- x |>
  group_by(Var1) |>
  arrange(desc(Var2)) |>
  mutate(prop = Freq / sum(Freq),
         n = sum(Freq)) %>%
  mutate(ypos = cumsum(prop)- 0.5*prop ) |>
  ungroup()

ext6d <- data |>
  dplyr::mutate(label = paste0("n=",Freq,"/",n,"\n",
                               scales::percent(prop)),
                label = ifelse(grepl("0.00", label), "", label),
                Var1 = gsub("_", " ", Var1),
                Var1 = factor(Var1, levels = c("epithelial hypoM",
                                               "distal epithelial hyperM",
                                               "immune hypoM",
                                               "proximal epithelial hyperM"))) |>
  ggplot(aes(x = 1,
             y= prop,
             fill = Var2)) +
  geom_bar(stat = "identity") +
  facet_wrap(~Var1, nrow = 2) +
  coord_polar(theta = "y",
  ) +
  ggrepel::geom_text_repel(aes(y = ypos,
                      label = label,
                      colour = Var2),
            size = 3,
            segment.colour = "black",
            min.segment.length = 0.2,
            nudge_x = 0.95
            ) +
  theme_void() +
  theme(legend.position = "top") +
  scale_fill_manual(values=cols2[c(1, 2, 3, 8)], name="study",
                    aesthetics = c("fill", "colour"))

plot <- (ext6a|ext6b)/(ext6c|ext6d) + plot_annotation(tag_levels = "a") + plot_layout(widths = c(0.25, 0.75))

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s6_pcgt.pdf"), width = 11, height = 13)
print(plot)
dev.off()
```

\includegraphics{1-figure-panels/s6_pcgt.pdf}

\textbf{Supplementary Figure 6. Polycomb group target enrichment in CpGs in the four sets, association with expression, and overlap with previously identified sites. a} Enrichment for Polycomb group target genes amongst CpGs in the different sets. \textbf{b} Histogram of p values for correlation of CpG loci with matched gene expression in TCGA-LUAD and TCGA-LUSC samples, based on genomic location of the CpG. \textbf{c} Density plot showing the association of CpG methylation with gene expression by genomic region. Only CpGs that were significantly correlated with expression after Bonferroni correction were included (p \textless 0.05, 55/98). \textbf{d} Overlap of CpGs with previously identified smoking CpGs in studies by Joehanes et al., Teschendorff et al., or Christiansen et al.


\newpage

\invisiblesection{Supplementary Figure 7. Epigenetically inferred cell type composition in datasets used in this study.}

```{r s7-epidish}
source(here("0-source", "expand_celltypes.R"))

# a smoking --------------------
load(here("1-analysis-pipeline/0-data/data.Rdata"))
disc_val <- data |>
  dplyr::mutate(dataset = ifelse(dataset == "vaping set", "e-cigarette set", dataset)) |>
  dplyr::filter(!(dataset == "discovery set" & smoking_history == "Ex-smoker"))

load(here("1-analysis-pipeline/4-datasets/1-output/snuff_tobacco.Rdata"))
snuff <- pheno |>
  dplyr::mutate(dataset = "moist snuff tobacco set",
                sampletype = "saliva",
                smoking_history = case_when(smoking_current == "Control" ~ "Non-smoker",
                                            smoking_current == "Moist Snuff User" ~ "Moist snuff tobacco user",
                                            smoking_current == "Cigarette Smoker" ~ "Smoker"))

data <- plyr::rbind.fill(disc_val, snuff) |>
  dplyr::mutate(smoking_history = factor(smoking_history, levels = c("Never smoker", "Non-smoker",
                                                     "Ex-smoker",
                                                     "Moist snuff tobacco user",
                                                     "e-cigarette user",
                                                     "Smoker")),
                dataset = factor(dataset, levels = c("discovery set",
                                                     "validation set",
                                                     "e-cigarette set",
                                                     "moist snuff tobacco set"))) |>
  expand_celltypes()


a <- data |>
  ggplot(aes(x = celltype,
             y = cell_prop,
             fill = smoking_history)) +
  geom_boxplot() +
  facet_wrap(dataset ~ sampletype,
             nrow = 2) +
  coord_flip() +
  xlab("") + ylab("Cell proportion") +
  theme_minimal() +
  theme(legend.position = "top", aspect.ratio = 1.2) +
  scale_fill_manual(values = cols2[c(7, 6, 3, 4, 9, 1)], name = "")

# b progression --------------------
load(here("1-analysis-pipeline/4-datasets/1-output/tcga.Rdata"))

matched <- tcga_data |>
  dplyr::group_by(barcode3) |>
  dplyr::count() |>
  dplyr::filter(n==2)

tcga <- tcga_data |>
  dplyr::filter(barcode3 %in% matched$barcode3) |>
  dplyr::mutate(dataset = project,
                type = ifelse(type == "Control", "Control lung tissue", "Cancerous lung tissue"))

load(here("1-analysis-pipeline/4-datasets/1-output/cis_progression.Rdata"))
cis <- pheno |>
  dplyr::mutate(dataset = "CIS set",
                type = case_when(type == "Control" ~ "Control lung tissue",
                                type == "Regression" ~ "Regressing CIS lesion",
                                type == "Progression" ~ "Progressing CIS lesion"))

data <- plyr::rbind.fill(tcga, cis) |>
  dplyr::mutate(dataset = factor(dataset, levels = c("TCGA-LUAD", "TCGA-LUSC", "CIS set")),
                type = factor(type, levels = c("Control lung tissue",
                                               "Regressing CIS lesion",
                                               "Progressing CIS lesion",
                                               "Cancerous lung tissue"))) |>
  expand_celltypes()

b <- data |>
  ggplot(aes(x = celltype,
             y = cell_prop,
             fill = type)) +
  geom_boxplot() +
  facet_wrap(~ dataset,
             nrow = 1) +
  coord_flip() +
  xlab("") + ylab("Cell proportion") +
  theme_minimal() +
  theme(legend.position = "top",
        , aspect.ratio = 1.2) +
  scale_fill_manual(values = cols2[c(9, 4, 3, 1)], name = "")


plot <- (a / b) + plot_layout(heights = c(2, 1)) + plot_annotation(tag_levels = "a")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s7_celltypes.pdf"),
                     width = 10, height = 11)
print(plot)
dev.off()
```

\includegraphics{1-figure-panels/s7_celltypes.pdf}

\textbf{Supplementary Figure 7. Epigenetically inferred cell type composition in datasets used in this study. a} Inferred cell type proportions in surrogate samples (buccal, blood, cervical) in various datasets included in this study. \textbf{b} Inferred cell type proportions in lung tissue datasets. Values were inferred using the EpiDISH package and reference matrices centEpiFibIC.m, and additionally, hierarchical EpiDISH (hEpiDISH) was applied using the centBloodSub.m reference matrix.

\textbf{Abbreviations}: TCGA, The Cancer Genome Atlas. LUAD, lung adenocarcinoma. LUSC, lung squamous cell carcinoma. CIS, carcinoma in situ.

\newpage

\invisiblesection{Supplementary Figure 8. Correction of mean beta values for cell type heterogeneity.}

```{r s8-correction}
load(here("1-analysis-pipeline/0-data/data.Rdata"))
dat <- data |>
  dplyr::filter(sampletype == "buccal" & dataset == "discovery set" & !is.na(smoking_history)) |>
  droplevels() |>
  dplyr::mutate(smoking_history = factor(smoking_history, levels = c("Never smoker", "Ex-smoker", "Smoker")),
                type = ifelse(smoking_history == "Never smoker", "Control", as.character(smoking_history)),
                type = factor(type, levels = c("Control", "Ex-smoker", "Smoker"))) |>
  tibble::rowid_to_column("id")

ext8a <- dat |>
  ggplot(aes(x = ic,
             y = WID_SMK450_proximal_epithelial_hyperM,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  # ggpmisc::stat_poly_eq(aes(label = paste(after_stat(eq.label), # print Eq
  #                                after_stat(rr.label), sep = "*\", \"*"))) +
  # facet_wrap(~set,nrow = 1,
  #            labeller = labeller(set = labs)) +
  scale_colour_manual(values = cols2[c(7, 3, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "top",
        strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8) +
  xlab("immune cell proportion") +
  ylab("mean β\n(uncorrected)")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext8a.pdf"), width = 5, height = 5)
print(ext8a)
dev.off()

dat2 <- dat |>
  correct_ic(correction = "internal") |>
  tidyr::pivot_longer(cols = c(WID_SMK450_proximal_epithelial_hyperM,
                               WID_SMK450_proximal_epithelial_hyperM_corr),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(ic = ifelse(score == "WID_SMK450_proximal_epithelial_hyperM_corr", 0, ic))

ext8b <- dat2 |>
  ggplot() +
  geom_point(aes(x = ic,
                 y = value,
                colour = smoking_history,

                shape = score),
               size = 1) +
  geom_line(aes(x = ic,
                y = value,
                group = id,
                colour = smoking_history),
            alpha = 0.3,
            linetype = "dotdash") +
  scale_colour_manual(values = cols2[c(7, 3, 1)], name = "") +
    scale_shape_manual(values = c(19, 2)) +
  theme_minimal() +
  theme(legend.position = "top",
        strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8) +
  xlab("immune cell proportion") +
  ylab("mean β\n(corrected)")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext8b.pdf"), width = 5, height = 5)
print(ext8b)
dev.off()

# Raw values
dat2 <- dat |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"),
                      names_to = "set",
                      values_to = "value") |>
  dplyr::mutate(set = gsub("WID_SMK450_", "", set),
                set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                               "immune hypoM",
                                               "distal epithelial hyperM",
                                               "proximal epithelial hyperM")))

ext8c <- dat2 |>
  ggplot(aes(x = ic,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1) +
  scale_colour_manual(values = cols2[c(7, 3, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8) +
  xlab("immune cell proportion") +
  ylab("mean β\n(uncorrected)")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext8c.pdf"), width = 12, height = 3.75)
print(ext8c)
dev.off()


# Corrected values
dat2 <- dat |>
  correct_ic(correction = "internal") |>
  tidyr::pivot_longer(cols = paste0(c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"), "_corr"),
                      names_to = "set",
                      values_to = "value") |>
  dplyr::mutate(set = gsub("WID_SMK450_|_corr", "", set),
                set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                               "immune hypoM",
                                               "distal epithelial hyperM",
                                               "proximal epithelial hyperM")))
ext8d <- dat2 |>
  ggplot(aes(x = ic,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5,
             shape = 2) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1) +
  scale_colour_manual(values = cols2[c(7, 3, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8) +
  xlab("immune cell proportion") +
  ylab("mean β\n(corrected)")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext8d.pdf"), width = 12, height = 3.75)
print(ext8d)
dev.off()
```

\includegraphics[trim = 0 10cm 0 0]{1-figure-panels/s8_correction.pdf}

\textbf{Supplementary Figure 8. Correction of mean beta values for cell type heterogeneity. a} Raw mean beta values of proximal epithelial hyperM CpGs (450k) in the discovery set buccal samples. The correction approach is based on slopes and residuals for never, ex- and current smokers by fitting individual linear models for each type. \textbf{b} Visualisation of corrected mean beta of values in a, projecting values to immune cell proportion = 0. \textbf{c} Raw mean methylation values of all four groups of CpGs in buccal samples in the discovery set. \textbf{d} Corrected mean methylation values of all four groups of CpGs in buccal samples in the discovery set. 

\newpage

\invisiblesection{Supplementary Figure 9. Correlation of corrected mean beta values with smoking pack years in the validation set (buccal and blood samples).}

```{r s9-spy}
load(here("1-analysis-pipeline/0-data/data.Rdata"))

# s figure 9a-------------------------
dat <- data |>
  dplyr::filter(dataset == "validation set" & sampletype == "buccal") |>
  droplevels() |>
  dplyr::mutate(smoking_history = factor(smoking_history, levels = c("Never smoker", "Ex-smoker", "Smoker"))) |>
  correct_ic(correction = "internal")
dat2 <- dat |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM_corr", "WID_SMK450_immune_hypoM_corr", "WID_SMK450_distal_epithelial_hyperM_corr", "WID_SMK450_proximal_epithelial_hyperM_corr"),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(score = gsub("WID_SMK450_|_corr", "", score),
                score = gsub("_", " ", score),
                score = factor(score, levels = c("epithelial hypoM",
                                                 "immune hypoM",
                                                 "distal epithelial hyperM",
                                                 "proximal epithelial hyperM")),
                smoking_history = factor(smoking_history, levels = c("Never smoker", "Ex-smoker", "Smoker")))

a <- dat2 |>
  filter(smoking_history != "Never smoker") |>
  ggplot(aes(x = spy,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~score,nrow = 1) +
  scale_colour_manual(values = cols2[c(3, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1,
        axis.title.y = element_markdown()) +
  ggpubr::stat_cor(size = 3,
                   method = "pearson",
                   show.legend = F) +
  xlab("smoking pack years") +
  ylab("mean β<br><span style='color:#224B5E;'><b>buccal samples</b></span></b><br>(corrected)")

# Fig b-------------------------
dat <- data |>
  dplyr::filter(dataset == "validation set" & sampletype == "blood") |>
  droplevels() |>
  dplyr::mutate(smoking_history = factor(smoking_history, levels = c("Never smoker", "Ex-smoker", "Smoker")))

dat2 <- dat |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(score = gsub("WID_SMK450_", "", score),
                score = gsub("_", " ", score),
                ic = 1-(hepidish_Mono+hepidish_Neutro+hepidish_Eosino),
                score = factor(score, levels = c("epithelial hypoM",
                                                 "immune hypoM",
                                                 "distal epithelial hyperM",
                                                 "proximal epithelial hyperM")),
                smoking_history = factor(smoking_history, levels = c("Never smoker", "Ex-smoker", "Smoker")))

b <- dat2 |>
  filter(smoking_history != "Never smoker") |>
  ggplot(aes(x = spy,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~score,nrow = 1) +
  scale_colour_manual(values = cols2[c(3, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1,
        axis.title.y = element_markdown()) +
  ggpubr::stat_cor(size = 3,
                   method = "pearson",
                   show.legend = F) +
  xlab("smoking pack years") +
  ylab("mean β<br><span style='color:#C0BE84;'><b>blood samples</b></span></b><br>(corrected)")

plot <- (a/b) + plot_annotation(tag_levels = "a")
grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s9_spy.pdf"), width = 9, height = 7)
print(plot)
dev.off()
```

\includegraphics{1-figure-panels/s9_spy.pdf}

\textbf{Supplementary Figure 9. Correlation of corrected mean beta values with smoking pack years (buccal and blood samples).} For buccal and blood samples, smoking pack year information was available. \textbf{a} Pearson correlation of corrected mean beta values of the four groups of CpGs with smoking pack years in buccal samples. \textbf{b} Pearson correlation of corrected mean beta values of the four groups of CpGs with smoking pack years in buccal samples.
 
\newpage

\invisiblesection{Supplementary Figure 10. E-cigarette dataset raw and corrected methylation values and correlation of corrected scores with cigarette and e-cigarette use.}

```{r s10-e-cig}
load(here("1-analysis-pipeline/0-data/data.Rdata"))
dat <- data |>
  dplyr::filter(dataset == "vaping set" & !is.na(smoking_history)) |>
  droplevels() |>
  dplyr::mutate(smoking_history = factor(smoking_history, levels = c("Never smoker", "e-cigarette user", "Smoker")),
                type = ifelse(smoking_history == "Never smoker", "Control", as.character(smoking_history)),
                type = factor(type, levels = c("Control", "e-cigarette user", "Smoker"))) |>
  tibble::rowid_to_column("id")

# EPIC scores
# Raw values - EPIC
dat2 <- dat |>
  tidyr::pivot_longer(cols = c("WID_SMK_epithelial_hypoM", "WID_SMK_immune_hypoM", "WID_SMK_distal_epithelial_hyperM", "WID_SMK_proximal_epithelial_hyperM"),
                      names_to = "set",
                      values_to = "value") |>
  dplyr::mutate(set = gsub("WID_SMK_", "", set),
                set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                            "immune hypoM",
                                               "distal epithelial hyperM",
                                               "proximal epithelial hyperM")),
                smoking_history = factor(smoking_history, levels = c("Never smoker", "e-cigarette user", "Smoker")))

a <- dat2 |>
  ggplot(aes(x = ic,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1, scales = "free_y") +
  scale_colour_manual(values = cols2[c(7, 9, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8,
        plot.subtitle = element_markdown()) +
  xlab("immune cell proportion") +
  ylab("mean β\n(raw)")

# Corrected values
dat2 <- dat |>
  correct_ic(correction = "internal") |>
  tidyr::pivot_longer(cols = paste0(c("WID_SMK_epithelial_hypoM", "WID_SMK_immune_hypoM", "WID_SMK_distal_epithelial_hyperM", "WID_SMK_proximal_epithelial_hyperM"), "_corr"),
                      names_to = "set",
                      values_to = "value") |>
  dplyr::mutate(set = gsub("WID_SMK_|_corr", "", set),
                set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                               "immune hypoM",
                                               "distal epithelial hyperM",
                                               "proximal epithelial hyperM")),
                smoking_history = factor(smoking_history, levels = c("Never smoker", "e-cigarette user", "Smoker")))

b <- dat2 |>
  ggplot(aes(x = ic,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5, shape = 2) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1, scales = "free_y") +
  scale_colour_manual(values = cols2[c(7, 9, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8,
        plot.subtitle = element_markdown()) +
  xlab("immune cell proportion") +
  ylab("mean β\n(corrected)")


load(here("1-analysis-pipeline/0-data/data.Rdata"))

c <-dat2 |>
  dplyr::filter(smoking_history == "e-cigarette user") |>
  ggplot(aes(x = vaping_smk_n,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1, scales = "free_y") +
  scale_colour_manual(values = cols2[c(8)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1) +
  ggpubr::stat_cor(size = 3,
                   method = "pearson",
                   show.legend = F) +
  xlab("cigarettes tried (ever)") +
  ylab("mean β\n(corrected)")

d <-dat2 |>
  dplyr::filter(smoking_history == "e-cigarette user") |>
  ggplot(aes(x = as.numeric(vaping_ml_day),
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1, scales = "free_y") +
  scale_colour_manual(values = cols2[c(8)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1) +
  ggpubr::stat_cor(size = 3,
                   method = "pearson",
                   show.legend = F) +
  xlab("vaping mL/day") +
  ylab("mean β\n(corrected)")

plot <- (a/b/c/d) + plot_annotation(tag_levels = "a")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s10_ecig.pdf"), width = 10, height = 12)
print(plot)
dev.off()
```

\includegraphics{1-figure-panels/s10_ecig.pdf}

\textbf{Supplementary Figure 11. E-cigarette dataset raw and corrected methylation values and correlation of corrected scores with cigarette and e-cigarette use. a} Raw methylation mean beta values for each group of CpGs. \textbf{b} Corrected methylation mean beta values in the same samples. \textbf{c} Correlation of methylation beta values and reported number of cigarettes tried (ever) by e-cigarette users. \textbf{d} Correlation of methylation beta values and reported mL/day use of e-cigarette liquid.

\newpage

\invisiblesection{Supplementary Figure 11. E-cigarette differential methylation at individual loci and gene set enrichment.}

```{r s11-ecig-heatmap}
# Load with smk cpgs
load(here("1-analysis-pipeline/2-output/WID_SMK_cpgs.Rdata"))

# Helper function for dbs
load_and_name <- function(path = "",
                          set = "",
                          type2 = ""){
  tmp <- readRDS(path) |>
    as.data.frame() |>
           tibble::rownames_to_column("cg") |>
    dplyr::filter(cg %in% WID_SMK_cpgs$cg) |>
           dplyr::mutate(set = set,
                         type2 = type2)

  return(tmp)
}

db_orig <- load_and_name(path = here("1-analysis-pipeline/1-delta-beta/1a-buccal/delta-beta.Rds"),
              set = "discovery",
              type2 = "smoker (discovery set)")

db_ecig <- load_and_name(path = here("1-analysis-pipeline/6-output/delta-beta.Rds"),
                         set = "ecig",
                         type2 = "e-cigarette user")

# Put together
dat <- plyr::rbind.fill(db_orig, db_ecig) |>
  tidyr::pivot_wider(id_cols = "cg",
                     names_from = c("set", "type2"),
                     values_from = c("db_immune", "db_epithelial")) |>
  tidyr::drop_na()
mat <- dat[,-1]
cgs <- dat$cg

colnames(mat) <- stringr::str_split(colnames(mat), "_", simplify = T)[,4]
mat2 <- t(mat)

x <- match(cgs, WID_SMK_cpgs$cg)
cg_sets <- WID_SMK_cpgs$set[x]
epi_imm  <- paste0(stringr::str_split(colnames(dat)[-1], "_", simplify = T)[,2], "\nfraction")


a <- Heatmap(mat2,
        column_split = cg_sets,
        row_split = epi_imm,
        show_column_names = F,
        show_column_dend = F,
        name = "∆ β",
        show_row_names = T,
        cluster_columns = T,
        cluster_rows = T,
        row_names_gp = grid::gpar(fontsize = 9),
        column_title_gp = grid::gpar(fontsize = 9),
        row_title_gp = grid::gpar(fontsize = 9),
        col = circlize::colorRamp2(breaks = seq(from = -0.2, to = 0.2,
                                                length.out = 9),
                                   colors = rev(cols2)))

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/ext11a.pdf"), width = 10, height = 2)
print(a)
dev.off()

# Enrichment ---------------
load(here("1-analysis-pipeline/7-output/epi_hypo_GO.Rdata"))
load(here("1-analysis-pipeline/7-output/prox_epi_hyper_GO.Rdata"))

b <- enrichplot::cnetplot(epi_hypo_react_all,
                     size = 3,
                     cex.params = list(category_label = 0.92,
                                       gene_label = 0.9),
                     color.params = list(category = cols[9],
                                         gene = "grey40"),
                     circular = F)

grDevices::pdf(here("2-markdown/1-figure-panels/ext11b.pdf"), width = 6, height = 7)
print(b)
dev.off()

c <- enrichplot::cnetplot(prox_epi_hyper_react_all,
                     size = 3,
                     cex.params = list(category_label = 0.92,
                                       gene_label = 0.9),
                     color.params = list(category = cols[1],
                                         gene = "grey40"),
                     circular = F)

grDevices::pdf(here("2-markdown/1-figure-panels/ext11c.pdf"), width = 6, height = 7)
print(c)
dev.off()
```

\includegraphics[trim = 0 12cm 0 0]{1-figure-panels/s11_ecig-enrich.pdf}


\textbf{Supplementary Figure 11. E-cigarette differential methylation at individual loci and gene set enrichment. a} Heatmap comparing delta beta values in the inferred immune and epithelial fractions in buccal samples of the discovery set and e-cigarette users. \textbf{b} Reactome pathway enrichment for sites sharing the same directionality in the epithelial fraction of epithelial hypoM as in smokers in the discovery set. \textbf{c} Reactome pathway enrichment for sites sharing the same directionality in the epithelial fraction of proximal epithelial hyperM as in smokers in the discovery set.


\invisiblesection{Supplementary Figure 12. Smokeless tobacco use dataset raw and corrected methylation values.}

```{r s12-smokeless}
load(here("1-analysis-pipeline/4-datasets/1-output/snuff_tobacco.Rdata"))

dat <- pheno |>
  dplyr::mutate(smoking_current = case_when(smoking_current == "Control" ~ "Non-smoker",
                                            smoking_current == "Cigarette Smoker" ~ "Smoker",
                                            smoking_current == "Moist Snuff User" ~ "Smokeless tobacco user"),
                smoking_history = smoking_current,
                smoking_history = factor(smoking_history, levels = c("Non-smoker", "Smokeless tobacco user", "Smoker"))) |>
  tibble::rowid_to_column("id")

dat2 <- dat |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"),
                      names_to = "set",
                      values_to = "value") |>
  dplyr::mutate(set = gsub("WID_SMK450_", "", set),
                set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                            "immune hypoM",
                                               "distal epithelial hyperM",
                                               "proximal epithelial hyperM")),
                smoking_history = factor(smoking_history, levels = c("Non-smoker", "Smokeless tobacco user", "Smoker")))

a <- dat2 |>
  ggplot(aes(x = ic,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1, scales = "free_y") +
  scale_colour_manual(values = cols2[c(7, 4, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "top",
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8,
        plot.subtitle = element_markdown()) +
  xlab("immune cell proportion") +
  ylab("mean β (raw)")

# Corrected values
dat2 <- dat |>
  correct_ic(correction = "internal") |>
  tidyr::pivot_longer(cols = paste0(c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"), "_corr"),
                      names_to = "set",
                      values_to = "value") |>
  dplyr::mutate(set = gsub("WID_SMK450_|_corr", "", set),
                set = gsub("_", " ", set),
                set = factor(set, levels = c("epithelial hypoM",
                                               "immune hypoM",
                                               "distal epithelial hyperM",
                                               "proximal epithelial hyperM")),
                smoking_history = factor(smoking_history, levels = c("Non-smoker", "Smokeless tobacco user", "Smoker")))

b <- dat2 |>
  ggplot(aes(x = ic,
             y = value,
             colour = smoking_history)) +
  geom_point(size = 0.5, shape = 2) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~set,nrow = 1, scales = "free_y") +
  scale_colour_manual(values = cols2[c(7, 4, 1)], name = "") +
  theme_minimal() +
  theme(legend.position = "none",
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8,
        plot.subtitle = element_markdown()) +
  xlab("immune cell proportion") +
  ylab("mean β (corrected)")

plot <- ((a)/(b)) + plot_annotation(tag_levels = "a")

grDevices::cairo_pdf(here("2-markdown/1-figure-panels/s12_smokeless-tobacco.pdf"), width = 10, height = 6)
print(plot)
dev.off()
```

\includegraphics{1-figure-panels/s12_smokeless-tobacco.pdf}

\textbf{Supplementary Figure 12. Smokeless tobacco use dataset raw and corrected methylation values. a} Raw methylation mean beta values for each set of CpGs. \textbf{b} Corrected methylation mean beta values in the same samples. 

\invisiblesection{Supplementary Figure 13. Dependence of mean methylation values on immune cell composition in cancer tissue and carcinoma in situ lesions.}

```{r s13-tcga-cis-ic}
load(here("1-analysis-pipeline/4-datasets/1-output/tcga.Rdata"))
#
# # Fig a - ic -----------------
# # isolate matched samples (same exposure only)
matched <- tcga_data |>
  dplyr::group_by(barcode3) |>
  dplyr::summarise(x = paste0(type, collapse = ",")) |>
  dplyr::filter(x %in% c("Control,Cancer", "Cancer,Control"))

tmp <- tcga_data |>
  dplyr::filter(barcode3 %in% matched$barcode3) |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(score = gsub("WID_SMK450_|", "", score),
                score = gsub("_", " ", score),
                score = factor(score, levels = c("epithelial hypoM",
                                                 "immune hypoM",
                                                 "distal epithelial hyperM",
                                                 "proximal epithelial hyperM")),
                type = case_when(type == "Cancer" ~ "Cancer tissue",
                                 type == "Control" ~ "Matched normal tissue"),
                type = factor(type, levels = c("Matched normal tissue",
                                               "Cancer tissue")))

a <- tmp |>
  ggplot(aes(x = ic,
             y = value,
             colour = type)) +
  geom_point() +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(project~score, nrow = 2) +
  xlab("ic") +
  ylab("mean β\n")+
  scale_colour_manual(values = cols2[c(9,1)], name = "",
                      aesthetics = c("color")) +
  theme_minimal() +
  theme(legend.position = "top",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1) +
  xlab("immune cell proportion") +
  ylab("mean β\n")

# Cervical
load(here("1-analysis-pipeline/4-datasets/1-output/cervical-cancer.Rdata"))

matched = pheno |>
  dplyr::group_by(id) |>
  dplyr::count(type) |>
  tidyr::pivot_wider(id_cols = id,
                     names_from = type,
                     values_from = n) |>
  dplyr::filter(Cancer == 1 & Control==1)

tmp <- pheno |>
  dplyr::filter(id %in% matched$id) |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(score = gsub("WID_SMK450_|", "", score),
                score = gsub("_", " ", score),
                score = factor(score, levels = c("epithelial hypoM",
                                                 "immune hypoM",
                                                 "distal epithelial hyperM",
                                                 "proximal epithelial hyperM")),
                type = case_when(type == "Cancer" ~ "Cancer tissue",
                                 type == "Control" ~ "Matched normal tissue"),
                type = factor(type, levels = c("Matched normal tissue",
                                               "Cancer tissue")))

b<- tmp |>
  ggplot(aes(x = ic,
             y = value,
             colour = type)) +
  geom_point() +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~score, nrow = 1) +
  xlab("ic") +
  ylab("mean β\n")+
  scale_colour_manual(values = cols2[c(9,1)], name = "",
                      aesthetics = c("color")) +
  theme_minimal() +
  theme(legend.position = "top",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1) +
  xlab("immune cell proportion") +
  ylab("mean β\n")

# Raw CIS values
load(here("1-analysis-pipeline/4-datasets/1-output/cis_progression.Rdata"))

tmp <- pheno |>
  tidyr::pivot_longer(cols = c("WID_SMK450_epithelial_hypoM", "WID_SMK450_immune_hypoM", "WID_SMK450_distal_epithelial_hyperM", "WID_SMK450_proximal_epithelial_hyperM"),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(score = gsub("WID_SMK450_|", "", score),
                score = gsub("_", " ", score),
                score = factor(score, levels = c("epithelial hypoM",
                                                 "immune hypoM",
                                                 "distal epithelial hyperM",
                                                 "proximal epithelial hyperM")),
                type = case_when(type == "Control" ~ "Control tissue",
                                 type == "Regression" ~ "Regressing CIS lesion",
                                 type == "Progression" ~ "Progressing CIS lesion"),
                type = factor(type, levels = c("Control tissue",
                                               "Regressing CIS lesion",
                                               "Progressing CIS lesion")))

c <- tmp |>
  ggplot(aes(x = ic,
             y = value,
             colour = type)) +
  geom_point(aes(shape = smoking_history),
             alpha = 0.8) +
  geom_smooth(method = "lm", se = F) +
  facet_wrap(~score, nrow = 1) +
  xlab("ic") +
  ylab("mean β\n")+
  scale_colour_manual(values = cols2[c(9,7,1)], name = "",
                      aesthetics = c("color")) +
  theme_minimal() +
  theme(legend.position = "top",
        # strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 1) +
  xlab("immune cell proportion") +
  ylab("mean β\n")

plot <- (a/b/c) + plot_annotation(tag_levels = "a") + plot_layout(heights = c(2, 1, 1))

grDevices::cairo_pdf(here(here("2-markdown/1-figure-panels/s13_tcga-ic.pdf")),
                     width = 10, height = 16)
print(plot)
dev.off()
```

\includegraphics[width=0.8\textwidth, trim=0 1cm 0 0]{1-figure-panels/s13_tcga-ic.pdf}

\textbf{Supplementary Figure 13. Dependence of mean methylation values on immune cell composition in cancer tissue and carcinoma in situ lesions.  a} Raw methylation mean beta values for each set of CpGs versus inferred immune cell proportion in TCGA-LUAD and TCGA-LUSC samples with matched control tissue. \textbf{b} Raw methylation mean beta values for each set of CpGs versus inferred immune cell proportion in cervical cancer versus normal control tissue. \textbf{c} Raw methylation mean beta values for each set of CpGs in control samples or regressing or progressing CIS lesions.

\newpage

\section{Supplementary Tables}

\invisiblesection{Supplementary Table 1. Sample overview}

\textbf{Supplementary Table 1. Sample overview of datasets used to investigate the impact of tobacco, e-cigarette use, or smokeless tobacco, on different cell types.} 

```{r stable1, eval=F}
load(here("1-analysis-pipeline/0-data/data.Rdata"))

disco <- data |>
  dplyr::filter(dataset %in% c("discovery set")) |>
  dplyr::group_by(sampletype) |>
  dplyr::reframe(sampletype = sampletype,
                 platform = "EPIC",
                 n = as.character(n()),
                 accession = case_when(sampletype == "buccal" ~  "EGAS00001005055",
                                       sampletype == "blood" ~ "EGAS00001005055",
                                       sampletype == "cervical" ~ "EGAS00001005055"),
                 exsmk = paste0(sum(smoking_history == "Ex-smoker"),
                                " (", (round(sum(smoking_history == "Ex-smoker")/n()*100, 1)), "%)"),
                 smk = paste0(sum(smoking_history == "Smoker"),
                              " (", (round(sum(smoking_history == "Smoker")/n()*100, 1)), "%)"),
                 nsmk = paste0(sum(smoking_history == "Never smoker"),
                               " (", (round(sum(smoking_history == "Never smoker")/n()*100, 1)), "%)"),
                 age = paste0(round(median(age),2),
                              " (", round(quantile(age, probs = 0.25, na.rm = T), 0), "-", round(quantile(age, probs = 0.75, na.rm = T),0), ")"),
                 female = paste0(n(), " (100%)")) |>
  dplyr::distinct() |>
  ungroup() |>
  tidyr::pivot_longer(platform:female,
                      names_to = "name",
                      values_to = paste0("discovery set")) |>
  dplyr::mutate(name = case_when(name == "exsmk" ~ "Ex-smoker",
                                 name == "n" ~ "n",
                                 name == "platform" ~ "Platform",
                                 name == "accession" ~ "Accession",
                                 name == "smk" ~ "Smoker",
                                 name == "nsmk" ~ "Never smoker",
                                 name == "age" ~ "Age",
                                 name == "female" ~ "Female (%)"))

val <- data |>
  dplyr::filter(dataset %in% c("validation set")) |>
  dplyr::group_by(sampletype) |>
  dplyr::reframe(sampletype = sampletype,
                 platform = ifelse(sampletype == "cervical", "EPIC", "450K"),
                 n = as.character(n()),
                 accession = ifelse(sampletype == "cervical", "EGAS00001005055", "N/A*"),
                 exsmk = paste0(sum(smoking_history == "Ex-smoker"),
                                " (", (round(sum(smoking_history == "Ex-smoker")/n()*100, 1)), "%)"),
                 smk = paste0(sum(smoking_history == "Smoker"),
                              " (", (round(sum(smoking_history == "Smoker")/n()*100, 1)), "%)"),
                 nsmk = paste0(sum(smoking_history == "Never smoker"),
                               " (", (round(sum(smoking_history == "Never smoker")/n()*100, 1)), "%)"),
                 female = paste0(n(), " (100%)"),
                 age = "55") |>
  dplyr::distinct() |>
  ungroup() |>
  tidyr::pivot_longer(platform:age,
                      names_to = "name",
                      values_to = "validation set") |>
  dplyr::mutate(name = case_when(name == "exsmk" ~ "Ex-smoker",
                                 name == "n" ~ "n",
                                 name == "platform" ~ "Platform",
                                 name == "accession" ~ "Accession",
                                 name == "age" ~ "Age",
                                 name == "smk" ~ "Smoker",
                                 name == "nsmk" ~ "Never smoker",
                                 name == "female" ~ "Female (%)"))


vaping <- data |>
  dplyr::filter(dataset %in% c("vaping set")) |>
  dplyr::group_by(sampletype) |>
  dplyr::reframe(sampletype = sampletype,
                 platform = "EPIC",
                 accession = "N/A*",
                 n = as.character(n()),
                 vaping = paste0(sum(smoking_history == "e-cigarette user"),
                                 " (", (round(sum(smoking_history == "e-cigarette user")/n()*100, 1)), "%)"),
                 smk = paste0(sum(smoking_history == "Smoker"),
                              " (", (round(sum(smoking_history == "Smoker")/n()*100, 1)), "%)"),
                 nsmk = paste0(sum(smoking_history == "Never smoker"),
                               " (", (round(sum(smoking_history == "Never smoker")/n()*100, 1)), "%)"),
                 female = paste0(sum(sex == "Female"),
                                 " (", (round(sum(sex == "Female")/n()*100, 1)), "%)"),
                 age = paste0(round(median(age, na.rm = T),2),
                              " (", round(quantile(age, probs = 0.25, na.rm = T), 0), "-", round(quantile(age, probs = 0.75, na.rm = T),0), ")")) |>
  dplyr::distinct() |>
  ungroup() |>
  tidyr::pivot_longer(platform:age,
                      names_to = "name",
                      values_to = "e-cigarette use set") |>
  dplyr::mutate(name = case_when(name == "vaping" ~ "e-cigarette user",
                                 name == "n" ~ "n",
                                 name == "platform" ~ "Platform",
                                 name == "accession" ~ "Accession",
                                 name == "smk" ~ "Smoker",
                                 name == "age" ~ "Age",
                                 name == "nsmk" ~ "Never smoker",
                                 name == "female" ~ "Female (%)"))

load(here("1-analysis-pipeline/4-datasets/1-output/snuff_tobacco.Rdata"))
snuff <- pheno |>
  dplyr::mutate(sampletype = "saliva") |>
  dplyr::group_by(sampletype) |>
  dplyr::reframe(sampletype = sampletype,
                 platform = "450K",
                 accession = "GSE94876",
                 n = as.character(n()),
                 snuff = paste0(sum(smoking_current == "Moist Snuff User"),
                                " (", (round(sum(smoking_current == "Smokeless Tobacco User")/n()*100, 1)), "%)"),
                 smk = paste0(sum(smoking_current == "Cigarette Smoker"),
                              " (", (round(sum(smoking_current == "Cigarette Smoker")/n()*100, 1)), "%)"),
                 nsmk = paste0(sum(smoking_current == "Control"),
                               " (", (round(sum(smoking_current == "Control")/n()*100, 1)), "%)"),
                 female = paste0("0 (0%)"),
                 age = paste0(round(median(age, na.rm = T),2),
                              " (", round(quantile(age, probs = 0.25, na.rm = T), 0), "-", round(quantile(age, probs = 0.75, na.rm = T),0), ")"),) |>
  dplyr::distinct() |>
  ungroup() |>
  tidyr::pivot_longer(platform:age,
                      names_to = "name",
                      values_to = "moist snuff tobacco use set") |>
  dplyr::mutate(name = case_when(name == "snuff" ~ "Smokeless Tobacco User",
                                 name == "n" ~ "n",
                                 name == "platform" ~ "Platform",
                                 name == "accession" ~ "Accession",
                                 name == "age" ~ "Age",
                                 name == "female" ~ "Female (%)",
                                 name == "smk" ~ "Smoker",
                                 name == "nsmk" ~ "Non-smoker")) |>
  dplyr::rename(`smokeless tobacco use set` = `moist snuff tobacco use set`)

tmp <- disco |>
  dplyr::full_join(val) |>
  dplyr::full_join(vaping) |>
  dplyr::full_join(snuff)

tmp$name <- gsub("e-cigarette user", "E-cigarette user", tmp$name)
tmp$sampletype <- paste0(tmp$sampletype, " sample")

head(tmp)
library(gt)
t1 <- tmp |>
  # arrange(name) |>
  # dplyr::filter(!is.na(name)) |>
  dplyr::mutate(name = factor(name, levels = c("Accession",
                                               "n",
                                               "Platform",
                                               "Female (%)",
                                               "Age",
                                               "Never smoker",
                                               "Non-smoker",
                                               "Ex-smoker",
                                               "Smokeless Tobacco User",
                                               "E-cigarette user",
                                               "Smoker"))) |>
  arrange(name) |>
  gt(groupname_col = "sampletype", rowname_col = "name") |>
  tab_options(row_group.font.weight = "bold",
              data_row.padding = 2.5,
              column_labels.font.size = 12,
              table.font.size = 10,
              row_group.padding = 5,
              table.width = px(650),
              table.font.names = "Helvetica",
              row_group.border.top.width = px(2),
              row_group.border.bottom.width = px(2),
              stub.border.width = px(0),
              heading.title.font.size = 12)  |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse((x %in% c("NA") | is.na(x)), "-", x))
                 }) |>
  fmt_markdown(columns = everything())  |>
  cols_align(align = "left",
             columns = everything()) |>
  tab_stub_indent(rows = everything(), indent = "increase") |>
  cols_label(
    `discovery set` = html("<b>discovery set</b><br><br>n=2,341"),
    `validation set` = html("<b>validation set</b><br><br>n=304"),
    `e-cigarette use set` = html("<b>e-cigarette<br>use set</b><br>n=350"),
    `smokeless tobacco use set` = html("<b>smokeless tobacco<br>use set</b><br>n=120")
  ) |> tab_style(
    style = cell_borders(
      weight = px(0)
    ),
    locations = cells_body(
      columns = everything(),
      rows = everything()
    )) |>
  tab_style(
    style = cell_borders(weight = px(0)),
    locations = cells_stub(rows = everything())
  ) |>
  opt_stylize(style = 2) |> 
  tab_footnote(footnote = html("* data not publicly deposited due to restrictions on informed consent."))

t1 |>
  gtsave(here("2-markdown/1-figure-panels/s-table-1.html"))

pagedown::chrome_print(here("2-markdown/1-figure-panels/s-table-1.html"), output = here("2-markdown/1-figure-panels/s-table-1.pdf"))
```

\includegraphics[trim = 0 5cm 0 0]{1-figure-panels/s-table-1.pdf}

\newpage

\invisiblesection{Supplementary Table 2. Overview of CpGs associated with hyper- or hypomethylation in smokers compared to never smokers in at least one of the tissues or cell types.}

```{r stable2}
load(here("1-analysis-pipeline/2-output/WID_SMK_cpgs.Rdata"))
load(here("1-analysis-pipeline/2-output/plot_matrix.Rdata"))

mat <- mat |>
  as.data.frame() |>
  tibble::rownames_to_column("cg")

table <- WID_SMK_cpgs |>
  dplyr::select(-c(umap1, umap2)) |>
  dplyr::left_join(mat)

table |>
  dplyr::arrange(chr, pos) |>
  writexl::write_xlsx(here("2-markdown/2-tables/Suppl_Table_2.xlsx"),col_names = T)
```

\textbf{Supplementary Table 2. Overview of CpGs associated with hyper or hypomethylation in smokers compared to never smokers in at least one of the tissues or cell types.}

See separate .xlsx file.


\invisiblesection{Supplementary Table 3. Gene ontology and pathway enrichment for epithelial hypoM.}

\textbf{Supplementary Table 3. Gene ontology and pathway enrichment for epithelial hypoM.}

See separate .xlsx file.

\invisiblesection{Supplementary Table 4. Gene ontology and pathway enrichment for immune hypoM.}

\textbf{Supplementary Table 4. Gene ontology and pathway enrichment for immune hypoM.}

See separate .xlsx file.

\invisiblesection{Supplementary Table 5. Gene ontology and pathway enrichment for distal epithelial hyperM.}

\textbf{Supplementary Table 5. Gene ontology and pathway enrichment for distal epithelial hyperM.}

See separate .xlsx file.

\invisiblesection{Supplementary Table 6. Gene ontology and pathway enrichment for proximal epithelial hyperM.}

\textbf{Supplementary Table 6. Gene ontology and pathway enrichment for proximal epithelial hyperM.}

See separate .xlsx file.

\invisiblesection{Supplementary Table 7. Association of CpG methylation with gene expression in matched TCGA-LUAD and LUSC data.}

\textbf{Supplementary Table 7. Association of CpG methylation with gene expression in matched TCGA-LUAD and LUSC data.}

See separate .xlsx file.


\invisiblesection{Supplementary Table 8. Gene ontology and pathway enrichment for sites overlapping between cigarette smokers and e-cigarette users in the epithelial hypoM set.}

\textbf{Supplementary Table 8. Gene ontology and pathway enrichment for sites overlapping between cigarette smokers and e-cigarette users in the epithelial hypoM set.}

See separate .xlsx file.

\invisiblesection{Supplementary Table 9. Gene ontology and pathway enrichment for sites overlapping between cigarette smokers and e-cigarette users in the proximal epithelial hyperM set.}

\textbf{Supplementary Table 9. Gene ontology and pathway enrichment for sites overlapping between cigarette smokers and e-cigarette users in the proximal epithelial hyperM set}

See separate .xlsx file.

\newpage

\invisiblesection{Supplementary Table 10. Population characteristics of the ESTHER Study samples.}

\textbf{Supplementary Table 10. Population characteristics of the ESTHER Study samples.}

```{r stable9,eval = F}
tmp <- readxl::read_xlsx(here("0-source/s-tab9.xlsx"))

t9 <- tmp |> 
  dplyr::relocate(Groups, Characteristics, Controls, `LC Cases`) |> 
      gt(groupname_col = "Groups", 
         rownames_to_stub = T) |>
  cols_label(`LC Cases` = html("<b>LC cases</b><br>n=90"),
              Controls = html("<b>Controls</b><br>n=1403"),
             Characteristics = html("<b>Characteristics<b><br>&nbsp;")) |> 
  tab_options(row_group.font.weight = "bold",
              data_row.padding = 2.5,
              column_labels.font.size = 12,
              table.font.size = 10,
              row_group.padding = 5,
              table.width = px(500),
              table.font.names = "Helvetica",
              row_group.border.top.width = px(2),
              row_group.border.bottom.width = px(2),
              stub.border.width = px(0),
              heading.title.font.size = 12)  |>
  opt_stylize(style = 2, color = 'blue') |> 
      tab_style(
        style = cell_text(color = "white"),
        locations = cells_stub(rows = everything())
      ) |> 
  tab_footnote(footnote = html("Data missing for 1 lung cancer case and 43 controls (2.9 % of total participants)"),
               locations = cells_row_groups(groups = 'Smoking status at sampling [n (%)]')) |> 
  tab_footnote(footnote = html("Data missing for 6 lung cancer cases and 135 controls (9.4 % of total participants)"),
               locations = cells_row_groups(groups = 'Smoking pack-years')) |> 
  tab_footnote(footnote = html("p < 2.2e-16, as assessed by Mann-Whitney test."),
               locations = cells_body(rows = Groups == 'Immune_hypoM score',
                                      columns = Characteristics)) |> 
  tab_source_note(html("<b>Abbreviations</b>: LC, lung cancer; n, number; SD, standard deviation."))


t9 |>
  gtsave(here("2-markdown/1-figure-panels/s-table-9.html"))

pagedown::chrome_print(here("2-markdown/1-figure-panels/s-table-9.html"), output = here("2-markdown/1-figure-panels/s-table-9.pdf"))
```

\includegraphics[trim = 0 8cm 0 0]{1-figure-panels/s-table-9.pdf}

\newpage

\invisiblesection{Supplementary Table 11. Odds ratios of methylation values in the ESTHER study samples.}

\textbf{Supplementary Table 11. Odds ratios of methylation values in the ESTHER study samples.}

```{r stable10, eval=F}
tmp <- readxl::read_xlsx(here("0-source/s-tab10.xlsx")) |> 
  janitor::clean_names()

t10 <- tmp |> 
  gt() |> 
  cols_label(lc_cases = html("<b>LC cases</b><br>(n)"),
            controls = html("<b>Controls</b><br>(n)"),
             or_95_percent_ci_d_3 = html("OR (95% CI)"),
             p_valued_4 = html("p value"),
             or_95_percent_ci_d_5 = html("OR (95% CI)"),
             p_valued_6 = html("p value"),
             or_95_percent_ci_d_7 = html("OR (95% CI)"),
             p_valued_8 = html("p value")) |>
  tab_spanner(columns = or_95_percent_ci_d_3:p_valued_4,
              label = html("<b>Model 1</b>")) |>
  tab_spanner(columns = or_95_percent_ci_d_5:p_valued_6,
              label = html("<b>Model 2</b>")) |>
  tab_spanner(columns = or_95_percent_ci_d_7:p_valued_8,
              label = html("<b>Model 3</b>")) |> 
  tab_footnote(locations = cells_column_spanners(spanners = "<b>Model 1</b>"),
               footnote = html("Model 1: without adjustment for any confounders."))|> 
  tab_footnote(locations = cells_column_spanners(spanners = "<b>Model 2</b>"),
               footnote = html("Model 2: model 1 plus adjustment for age and sex."))|> 
  tab_footnote(locations = cells_column_spanners(spanners = "<b>Model 3</b>"),
               footnote = html("Model 3:  model 2 plus adjustment for smoking status and pack-years.")) |> 
  tab_style(style = cell_text(weight = 'bold'),
            locations = cells_body(columns = c(p_valued_4, p_valued_6, p_valued_8),
                                   rows = everything())) |> 
  tab_footnote(locations = cells_column_labels(columns = c(or_95_percent_ci_d_3,or_95_percent_ci_d_5, or_95_percent_ci_d_7,
                                                           p_valued_4,p_valued_6, p_valued_8)),
               footnote = "OR per standard deviation increase in the Immune_hypoM score; OR, 95% CI and two-sided p-values were generated from logistic regression model.") |> 
  tab_options(row_group.font.weight = "bold",
              data_row.padding = 2.5,
              column_labels.font.size = 12,
              table.font.size = 10,
              row_group.padding = 5,
              table.width = px(700),
              table.font.names = "Helvetica",
              row_group.border.top.width = px(2),
              row_group.border.bottom.width = px(2),
              stub.border.width = px(0),
              heading.title.font.size = 12)  |>
  opt_stylize(style = 2, color = 'blue') |> 
      tab_style(
        style = cell_text(color = "white"),
        locations = cells_stub(rows = everything())
      ) |> 
  tab_source_note(html("<b>Abbreviations</b>: CI, confidence interval; LC, lung cancer; n, number; OR, odds ratio."))

t10 |>
  gtsave(here("2-markdown/1-figure-panels/s-table-10.html"))

pagedown::chrome_print(here("2-markdown/1-figure-panels/s-table-10.html"), output = here("2-markdown/1-figure-panels/s-table-10.pdf"))
```

\includegraphics[trim = 0 5cm 0 0]{1-figure-panels/s-table-10.pdf}

\newpage

\section{Supplementary Movie}

\invisiblesection{Supplementary Movie 1. Illustration of immune cell correction algorithm.}

\textbf{Supplementary Movie 1. Illustration of immune cell correction algorithm.} Data is derived from Supplementary Figure 8a, b.

```{r smovie1}
library(gganimate)
library(av)

load(here("1-analysis-pipeline/0-data/data.Rdata"))
source(here("0-source/correct_ic.R"))

dat <- data |>
  dplyr::filter(sampletype == "buccal" & dataset == "discovery set" & !is.na(smoking_history)) |>
  droplevels() |>
  dplyr::mutate(smoking_history = factor(smoking_history, levels = c("Never smoker", "Ex-smoker", "Smoker")),
                type = ifelse(smoking_history == "Never smoker", "Control", as.character(smoking_history)),
                type = factor(type, levels = c("Control", "Ex-smoker", "Smoker"))) |>
  tibble::rowid_to_column("id")

data_animation <- dat |>
  correct_ic(correction = "internal") |>
  tidyr::pivot_longer(cols = c(WID_SMK450_proximal_epithelial_hyperM,
                               WID_SMK450_proximal_epithelial_hyperM_corr),
                      names_to = "score",
                      values_to = "value") |>
  dplyr::mutate(ic = ifelse(score == "WID_SMK450_proximal_epithelial_hyperM_corr", 0, ic)) |> 
  dplyr::mutate(animation = ifelse(grepl("corr", score), "2", "1"),
                score = gsub("_corr", "", score)) |>
  dplyr::select(type, score, value, ic, animation)
# head(data_animation)

plot <- data_animation |> 
  ggplot(aes(x = ic,
             y = value,
             colour = type)) +
  geom_point() +
  scale_colour_manual(values = cols2[c(7, 3, 1)], name = "") +
  scale_shape_manual(values = c(19, 2)) +
  theme_minimal() +
  theme(legend.position = "top",
        strip.background = element_rect(fill = cols2[5]),
        panel.border = element_rect(colour = "grey60",
                                    fill = NA),
        aspect.ratio = 0.8) +
  xlab("immune cell proportion") +
  ylab("mean β\n(corrected)") +
  # Here comes the gganimate code
  transition_states(
    animation,
    transition_length = 2,
    state_length = 1
  ) +
  enter_fade() + 
  exit_shrink() +
  ease_aes('sine-in-out')

# # anim_save(plot, file = '2-markdown/1-figure-panels/animation.gif')

# # save animations:
# b <- animate(plot, duration = 20, fps = 20, renderer = av_renderer())
# anim_save(here("2-markdown/1-figure-panels/animation.mp4"), b)
```